src/functions/huberLoss.jl

```diff
@@ -33,41 +33,41 @@ end
 is_convex(f::HuberLoss) = true
 is_smooth(f::HuberLoss) = true
 
-HuberLoss(rho::R=1.0, mu::R=1.0) where {R <: Real} = HuberLoss{R}(rho, mu)
+HuberLoss(rho::R=1, mu::S=1) where {R <: Real, S <: Real} = HuberLoss{R, S}(rho, mu)
 
-function (f::HuberLoss)(x::AbstractArray{T}) where T <: Union{Real, Complex}
+function (f::HuberLoss)(x::AbstractArray{T}) where {R <: Real, T <: Union{R, Complex{R}}}
     normx = norm(x)
     if normx <= f.rho
-        return (f.mu/2)*normx^2
+        return f.mu / R(2) * normx^2
     else
-        return f.rho*f.mu*(normx-f.rho/2)
+        return f.rho * f.mu * (normx - f.rho / R(2))
     end
 end
 
-function gradient!(y::AbstractArray{T}, f::HuberLoss, x::AbstractArray{T}) where T <: Union{Real, Complex}
+function gradient!(y::AbstractArray{T}, f::HuberLoss, x::AbstractArray{T}) where {R <: Real, T <: Union{R, Complex{R}}}
```

Comment:
`T<:RealOrComplex{R}`

Reply:
Ah! So I’ve actually changed my mind about that: `RealOrComplex{R}` is 17 characters while `Union{R, Complex{R}}` is 20, not much of a shorthand plus the latter is straight from `Base` and fully explicit